cli/tests/testdata/compile/dynamic_imports/main.js

```diff
@@ -0,0 +1,6 @@
+console.log("Starting the main module");
+
+setTimeout(() => {
+  console.log("Dynamic importing");
+  import("./import1.js").then(() => console.log("Dynamic import done."));
```

Comment:
This is statically analyzable by deno_graph and should already be in the module graph. I don't think we need to require people to specify a CLI flag for this and I think it should just be done automatically?

Reply:
It is not currently being added to the module graph, and removing the CLI flag makes the test fail. But even if we can detect uses of dynamic import (and the `Worker` constructor, because of #17657) with string arguments, a CLI flag would still be useful for cases that can't be detected.

Reply:
Should we open an issue about deno_graph not apparently including statically analysable dynamic imports into the module graph?

Reply:
It's there. We just have to make it work.

Reply:
I investigated this and it's an eszip thing. Opened https://github.com/denoland/eszip/pull/115

Once that's merged then the dynamic import test that's statically analyzable doesn't need the flag, so let's wait for that to merge and I'll do an eszip release then we can bump eszip in this PR.